app/components/post_draft/send_handler/__tests__/send_handler.test.tsx

```diff
@@ -0,0 +1,198 @@
+        postPriority: {
+            priority: 'urgent',
+            requested_ack: false,
+            persistent_notifications: false,
+        } as PostPriority,
+    };
+
+    beforeAll(async () => {
+        const server = await TestHelper.setupServerDatabase();
+        database = server.database;
+    });
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should match snapshot', () => {
+        const wrapper = renderWithEverything(
+            <SendHandler {...baseProps}/>, {database},
+        );
+        expect(wrapper.toJSON()).toBeTruthy();
+    });
+
+    it('should render DraftInput when not from draft view', () => {
+        const wrapper = renderWithEverything(
+            <SendHandler {...baseProps}/>, {database},
+        );
+        expect(wrapper.getByTestId('test_send_handler')).toBeTruthy();
+    });
+
+    it('should render SendDraft when from draft view', () => {
+        const props = {
+            ...baseProps,
+            isFromDraftView: true,
+            draftType: DRAFT_TYPE_DRAFT as DraftType,
+        };
+        const wrapper = renderWithEverything(
+            <SendHandler {...props}/>, {database},
+        );
+        expect(wrapper.getByTestId('send_draft_button')).toBeTruthy();
+        expect(wrapper.getByText('Send draft')).toBeTruthy();
+    });
+
+    it('should render Send text when draft type is scheduled', () => {
+        const props = {
+            ...baseProps,
+            isFromDraftView: true,
+            draftType: DRAFT_TYPE_SCHEDULED as DraftType,
+        };
+        const wrapper = renderWithEverything(
+            <SendHandler {...props}/>, {database},
+        );
+        expect(wrapper.getByTestId('send_draft_button')).toBeTruthy();
+        expect(wrapper.getByText('Send')).toBeTruthy();
+    });
+
+    it('should handle post priority updates', async () => {
+        const wrapper = renderWithEverything(
+            <SendHandler
+                {...baseProps}
+                canShowPostPriority={true}
+            />, {database},
+        );
+
+        const draftInput = wrapper.getByTestId('test_send_handler');
+        expect(draftInput).toBeTruthy();
+        expect(baseProps.postPriority.priority).toBe('urgent');
+        await waitFor(() => {
+            expect(wrapper.toJSON()).toBeTruthy();
+        });
+    });
+
+    it('should pass correct props to SendDraft when in draft view', () => {
+        const props = {
+            ...baseProps,
+            isFromDraftView: true,
+            draftType: DRAFT_TYPE_DRAFT as DraftType,
+            channelDisplayName: 'Test Channel',
+            draftReceiverUserName: 'test-user',
+            postId: 'test-post-id',
+        };
+
+        const wrapper = renderWithEverything(
+            <SendHandler {...props}/>, {database},
+        );
+
+        const sendDraftButton = wrapper.getByTestId('send_draft_button');
+        expect(sendDraftButton).toBeTruthy();
```

Comment:
Not sure how this translate to "should pass correct props".

Reply:
Updated the test.